eth/vm/computation.py

```diff
@@ -92,7 +92,86 @@ def memory_gas_cost(size_in_bytes: int) -> int:
     return total_cost
 
 
-class BaseComputation(Configurable, ABC):
+class BaseStackManipulation:
+    def stack_pop_ints(self, num_items: int) -> Tuple[int, ...]:
+        """
+        Pop and return a tuple of integers of length ``num_items`` from the stack.
+
+        Raise `eth.exceptions.InsufficientStack` if there are not enough items on
+        the stack.
+
+        Items are ordered with the top of the stack as the first item in the tuple.
+        """
+        raise Exception("Must be overridden by subclass, maybe at init time")
+
+    def stack_pop_bytes(self, num_items: int) -> Tuple[bytes, ...]:
+        """
+        Pop and return a tuple of bytes of length ``num_items`` from the stack.
+
+        Raise `eth.exceptions.InsufficientStack` if there are not enough items on
+        the stack.
+
+        Items are ordered with the top of the stack as the first item in the tuple.
+        """
+        raise Exception("Must be overridden by subclass, maybe at init time")
+
+    def stack_pop_any(self, num_items: int) -> Tuple[Union[int, bytes], ...]:
+        """
+        Pop and return a tuple of items of length ``num_items`` from the stack.
+        The type of each element will be int or bytes, depending on whether it was
+        pushed with stack_push_bytes or stack_push_int.
+
+        Raise `eth.exceptions.InsufficientStack` if there are not enough items on
+        the stack.
+
+        Items are ordered with the top of the stack as the first item in the tuple.
+        """
+        raise Exception("Must be overridden by subclass, maybe at init time")
+
+    def stack_pop1_int(self) -> int:
+        """
+        Pop and return an integer from the stack.
+
+        Raise `eth.exceptions.InsufficientStack` if the stack was empty.
+        """
+        raise Exception("Must be overridden by subclass, maybe at init time")
```

Comment:
Why this instead of using an `ABC` + `abstractmethod`?

Reply:
Because `abstractmethod` doesn't recognize methods overridden in `__init__` by assignment.